tictactoe.py

```diff
@@ -36,6 +36,11 @@ def check_row(self, y, player):
                 return False
         return True
 
+    def check_underscore(self, player, row, col):
```

Comment:
A better name is 'check_empty_square' or something similar. 

The fact that you're using an underscore to represent an empty square should be an implementation detail rather than part of the public API (if you just picked up this board class from somewhere, it might not be obvious that you'd use 'check_underscore' to check if a player already made a move, but it would be obvious that you could use check_empty_square or check_blank or check_location).

Another option here is to remove this method altogether and just return False from the make_move function when the move couldn't be made and True when it could (and adjust the playTicTacToe code).